src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpCodeVisitor.cs

```diff
@@ -116,6 +117,45 @@ protected override void Visit(TemplateChunk chunk)
             Writer.WriteEndMethodInvocation(false).WriteLine();
         }
 
+        protected override void Visit(ParentLiteralChunk chunk)
+        {
+            if (Context.Host.DesignTimeMode)
+            {
+                // Skip generating the chunk if we're in design time or if the chunk is empty.
+                return;
+            }
+            
+            if (Context.Host.EnableInstrumentation)
+            {
+                var start = chunk.Children[0].Association.Start.AbsoluteIndex;
```

Comment:
Did you find that the `chunk.Start` was different from the `chunk.Children[0].Association.Start`? You set it [here](https://github.com/aspnet/Razor/pull/613/files?diff=split#diff-60b1e905fa1bc92ee8f36cc9963e8b83R81).